pkg/nvlib/device/device.go

```diff
@@ -208,6 +209,33 @@ func (d *device) IsMigEnabled() (bool, error) {
 	return (mode == nvml.DEVICE_MIG_ENABLE), nil
 }
 
+// HasGpuFabricInfo() checks if a device has IMEX capabilities.
+func (d *device) HasGpuFabricInfo() (bool, error) {
+	if d.lib.hasSymbol("nvmlDeviceGetGpuFabricInfo") {
+		_, ret := d.GetGpuFabricInfo()
+		if ret == nvml.ERROR_NOT_SUPPORTED {
+			return false, nil
+		}
+		if ret != nvml.SUCCESS {
+			return false, fmt.Errorf("error getting GPU Fabric Info: %v", ret)
+		}
+		return true, nil
+	}
+
+	if d.lib.hasSymbol("nvmlDeviceGetGpuFabricInfoV") {
+		_, ret := d.GetGpuFabricInfoV().V2()
+		if ret == nvml.ERROR_NOT_SUPPORTED {
+			return false, nil
+		}
+		if ret != nvml.SUCCESS {
+			return false, fmt.Errorf("error getting GPU Fabric Info: %v", ret)
+		}
+		return true, nil
+	}
+
```

Comment:
I don't think this is necessary. It wouldn't have this one without the other.

Reply:
The previous one is marked as deprecated, this simply protect us in case we land in  and old driver machine, or a new driver machine.

Reply:
OK; they shouldn't ever remove it, but I'm OK including both.